src/status_im/components/toolbar_new/view.cljs

```diff
@@ -1,33 +1,114 @@
             [status-im.components.styles :as st]
-            [status-im.components.context-menu :refer [context-menu]]
+            [status-im.components.context-menu :as context-menu]
             [status-im.components.toolbar-new.actions :as act]
             [status-im.components.toolbar-new.styles :as tst]
-            [status-im.components.icons.vector-icons :as vi]
-            [reagent.core :as r]))
+            [status-im.components.icons.vector-icons :as vi]))
 
-(defn vec-icon [icon]
-  ((comp vec flatten vector) vi/icon icon))
+;; Navigation item
 
-(defn nav-button
-  [{:keys [handler accessibility-label image icon]}]
+(defn nav-item
+  [{:keys [handler accessibility-label style] :or {handler #(rf/dispatch [:navigate-back])}} item]
   [rn/touchable-highlight
-   (merge {:style    tst/toolbar-button
-           :on-press handler}
+   (merge {:on-press handler}
           (when accessibility-label
             {:accessibility-label accessibility-label}))
-   [rn/view
-    (if icon
-      [vec-icon icon]
-      [rn/image image])]])
+   [rn/view{:style style}
+    item]])
+
+(defn nav-button
+  [{:keys [icon icon-opts] :as props}]
+  [nav-item (merge {:style tst/nav-item-button} props)
+   [vi/icon icon icon-opts]])
+
+(defn nav-text
+  ([text] (nav-text text nil))
+  ([text handler] (nav-text nil text handler))
+  ([props text handler]
+   [rn/text (merge {:style tst/item-text :on-press (or handler #(rf/dispatch [:navigate-back]))})
+    text]))
+
+(defn nav-clear-text
+  ([text] (nav-clear-text text nil))
+  ([text handler]
+   (nav-text tst/item-text-white-background text handler)))
+
+(def default-nav-back [nav-button act/default-back])
+
+;; Content
+
+(defn content-title
+  ([title] (content-title nil title))
+  ([title-style title]
+   [rn/view {:style tst/toolbar-title-container}
+    [rn/text {:style (merge tst/toolbar-title-text title-style)
+              :font  :toolbar-title}
+     title]]))
+
+;; Actions
 
 (defn text-action [handler title]
-  [rn/text {:style tst/toolbar-right-action :onPress handler}
+  [rn/text {:style tst/toolbar-right-text-action :on-press handler}
    title])
 
+(def blank-action [rn/view tst/toolbar-action])
+
+(defn- option-actions [icon icon-opts options]
+  [context-menu/context-menu
+   [rn/view tst/toolbar-action
+    [vi/icon icon icon-opts]]
+   options
+   nil
+   tst/item])
+
+(defn- icon-action [icon icon-opts handler]
+  [rn/touchable-highlight {:style    tst/item
+                           :on-press handler}
+   [rn/view {:style tst/toolbar-action}
+    [vi/icon icon icon-opts]]])
+
+(defn actions [v]
+  [rn/view {:style tst/toolbar-actions}
+   (for [{:keys [image icon icon-opts options handler]} v]
+     (with-meta
+       (cond (= image :blank)
+             blank-action
+
+             options
+             [option-actions icon icon-opts options]
+
+             :else
+             [icon-action icon icon-opts handler])
+       {:key (str "action-" image)}))])
+
+(defn toolbar2
```

Comment:
What is toolbar vs toolbar2? Can we give a more descriptive name or a comment that explains when we use on over the other, and what the main difference is?

Reply:
`toolbar2` is the more general `toolbar`. I have not yet migrated everything to `toolbar2` but it will replace `toolbar`

Reply:
we have toolbar.toolbar, toolbar_new.toolbar and toolbar_new.toolbar2 :)

Reply:
TooManyToolbarsExceptions.

`toolbar2` should disappear soon.

Reply:
Can we document this with DEPRECATED and a short comment?

Reply:
sure